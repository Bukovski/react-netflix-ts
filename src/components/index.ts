export { default as Jumbotron } from './jumbotron/jumbotron.component';
export { default as Footer } from './footer/footer.component';
export { default as Accordion } from './accordion/accordion.component';
export { default as OptForm } from './opt-form/opt-form.component';
export { default as Feature } from './feature/feature.component';
export { default as Header } from './header/header.component';
export { default as Form } from './form/form.component';
export { default as Profiles } from './profiles/profile.component';
export { default as Loading } from './loading/loading.component';
export { default as Card } from './card/card.component';
export { default as Player } from './player/player.component';
