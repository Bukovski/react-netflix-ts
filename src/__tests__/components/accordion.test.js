import React from 'react';
import { render, fireEvent, screen, queryByAltText } from '@testing-library/react';
import faqsData from '../../fixtures/faqs';
import { Accordion } from '../../components';


describe('<Accordion />', () => {
  it('renders the <Accordion /> with populated data', () => {
    const { container, getByText } = render(
      <Accordion>
        <Accordion.Title>Frequently Asked Questions</Accordion.Title>
        <Accordion.Frame>
          {
            faqsData.map((item) => (
              <Accordion.Item key={item.id}>
                <Accordion.Header>{item.header}</Accordion.Header>
                <Accordion.Body>{item.body}</Accordion.Body>
              </Accordion.Item>
            ))
          }
        </Accordion.Frame>
      </Accordion>
    );
    
    expect(getByText('Frequently Asked Questions')).toBeTruthy();
    expect(getByText('What is Netflix?')).toBeTruthy();
    expect(getByText('How much does Netflix cost?')).toBeTruthy();
    expect(getByText('Where can I watch?')).toBeTruthy();
    expect(getByText('How do I cancel?')).toBeTruthy();
    expect(getByText('What can I watch on Netflix?')).toBeTruthy();
    
    expect(container.firstChild).toMatchSnapshot();
  });
  
  it('opens and closes the <Accordion /> component', () => {
    const { container, queryByText } = render(
      <Accordion>
        <Accordion.Title>Frequently Asked Questions</Accordion.Title>
        <Accordion.Frame>
          {
            faqsData.map((item) => (
              <Accordion.Item key={item.id}>
                <Accordion.Header>{item.header}</Accordion.Header>
                <Accordion.Body data-testid="accordion-body">{item.body}</Accordion.Body>
              </Accordion.Item>
            ))
          }
        </Accordion.Frame>
      </Accordion>
    );
    
    const linkOnParentDiv = queryByText(/Netflix is a streaming service that offers/).closest("div");
    const linkOnButton = queryByText('What is Netflix?');
    

    expect(linkOnParentDiv).toHaveClass("closed");
    expect(linkOnButton.querySelector('img[alt="Open"]')).toBeTruthy();
    
    fireEvent.click(linkOnButton);
    expect(linkOnParentDiv).not.toHaveClass("closed");
    expect(linkOnButton.querySelector('img[alt="Close"]')).toBeTruthy();
    
    fireEvent.click(linkOnButton);
    expect(linkOnParentDiv).toHaveClass("closed");
    expect(linkOnButton.querySelector('img[alt="Open"]')).toBeTruthy();
    
    expect(container.firstChild).toMatchSnapshot();
  });
});
