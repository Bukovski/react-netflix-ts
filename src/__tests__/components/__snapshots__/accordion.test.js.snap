// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> opens and closes the <Accordion /> component 1`] = `
<div
  class="accordionstyle__Container-sc-1ev5kio-0 dojqWO"
>
  <div
    class="accordionstyle__Inner-sc-1ev5kio-2 lltJcS"
  >
    <h1
      class="accordionstyle__Title-sc-1ev5kio-3 kOTkPD"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordionstyle__Frame-sc-1ev5kio-1 fbTwSn"
    >
      <div
        class="accordionstyle__Item-sc-1ev5kio-4 btjFtp"
      >
        <div
          class="accordionstyle__Header-sc-1ev5kio-5 iBTLgr"
        >
          What is Netflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
        <div
          class="accordionstyle__Body-sc-1ev5kio-6 jTyKSn closed"
          data-testid="accordion-body"
        >
          <span>
            Netflix is a streaming service that offers a wide variety of award-winning TV programmes, films, anime, documentaries and more – on thousands of internet-connected devices.

You can watch as much as you want, whenever you want, without a single advert – all for one low monthly price. There's always something new to discover, and new TV programmes and films are added every week!
          </span>
        </div>
      </div>
      <div
        class="accordionstyle__Item-sc-1ev5kio-4 btjFtp"
      >
        <div
          class="accordionstyle__Header-sc-1ev5kio-5 iBTLgr"
        >
          How much does Netflix cost?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
        <div
          class="accordionstyle__Body-sc-1ev5kio-6 jTyKSn closed"
          data-testid="accordion-body"
        >
          <span>
            Watch Netflix on your smartphone, tablet, smart TV, laptop or streaming device, all for one low fixed monthly fee. Plans start from £5.99 a month. No extra costs or contracts.
          </span>
        </div>
      </div>
      <div
        class="accordionstyle__Item-sc-1ev5kio-4 btjFtp"
      >
        <div
          class="accordionstyle__Header-sc-1ev5kio-5 iBTLgr"
        >
          Where can I watch?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
        <div
          class="accordionstyle__Body-sc-1ev5kio-6 jTyKSn closed"
          data-testid="accordion-body"
        >
          <span>
            Watch anywhere, anytime, on an unlimited number of devices. Sign in with your Netflix account to watch instantly on the web at netflix.com from your personal computer or on any internet-connected device that offers the Netflix app, including smart TVs, smartphones, tablets, streaming media players and game consoles.

You can also download your favourite programmes with the iOS, Android, or Windows 10 app. Use downloads to watch while you're on the go and without an internet connection. Take Netflix with you anywhere.
          </span>
        </div>
      </div>
      <div
        class="accordionstyle__Item-sc-1ev5kio-4 btjFtp"
      >
        <div
          class="accordionstyle__Header-sc-1ev5kio-5 iBTLgr"
        >
          How do I cancel?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
        <div
          class="accordionstyle__Body-sc-1ev5kio-6 jTyKSn closed"
          data-testid="accordion-body"
        >
          <span>
            Netflix is flexible. There are no annoying contracts and no commitments. You can easily cancel your account online in two clicks. There are no cancellation fees – start or stop your account at any time.
          </span>
        </div>
      </div>
      <div
        class="accordionstyle__Item-sc-1ev5kio-4 btjFtp"
      >
        <div
          class="accordionstyle__Header-sc-1ev5kio-5 iBTLgr"
        >
          What can I watch on Netflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
        <div
          class="accordionstyle__Body-sc-1ev5kio-6 jTyKSn closed"
          data-testid="accordion-body"
        >
          <span>
            Netflix has an extensive library of feature films, documentaries, TV programmes, anime, award-winning Netflix originals, and more. Watch as much as you want, any time you want.
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> renders the <Accordion /> with populated data 1`] = `
<div
  class="accordionstyle__Container-sc-1ev5kio-0 dojqWO"
>
  <div
    class="accordionstyle__Inner-sc-1ev5kio-2 lltJcS"
  >
    <h1
      class="accordionstyle__Title-sc-1ev5kio-3 kOTkPD"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordionstyle__Frame-sc-1ev5kio-1 fbTwSn"
    >
      <div
        class="accordionstyle__Item-sc-1ev5kio-4 btjFtp"
      >
        <div
          class="accordionstyle__Header-sc-1ev5kio-5 iBTLgr"
        >
          What is Netflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
        <div
          class="accordionstyle__Body-sc-1ev5kio-6 jTyKSn closed"
        >
          <span>
            Netflix is a streaming service that offers a wide variety of award-winning TV programmes, films, anime, documentaries and more – on thousands of internet-connected devices.

You can watch as much as you want, whenever you want, without a single advert – all for one low monthly price. There's always something new to discover, and new TV programmes and films are added every week!
          </span>
        </div>
      </div>
      <div
        class="accordionstyle__Item-sc-1ev5kio-4 btjFtp"
      >
        <div
          class="accordionstyle__Header-sc-1ev5kio-5 iBTLgr"
        >
          How much does Netflix cost?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
        <div
          class="accordionstyle__Body-sc-1ev5kio-6 jTyKSn closed"
        >
          <span>
            Watch Netflix on your smartphone, tablet, smart TV, laptop or streaming device, all for one low fixed monthly fee. Plans start from £5.99 a month. No extra costs or contracts.
          </span>
        </div>
      </div>
      <div
        class="accordionstyle__Item-sc-1ev5kio-4 btjFtp"
      >
        <div
          class="accordionstyle__Header-sc-1ev5kio-5 iBTLgr"
        >
          Where can I watch?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
        <div
          class="accordionstyle__Body-sc-1ev5kio-6 jTyKSn closed"
        >
          <span>
            Watch anywhere, anytime, on an unlimited number of devices. Sign in with your Netflix account to watch instantly on the web at netflix.com from your personal computer or on any internet-connected device that offers the Netflix app, including smart TVs, smartphones, tablets, streaming media players and game consoles.

You can also download your favourite programmes with the iOS, Android, or Windows 10 app. Use downloads to watch while you're on the go and without an internet connection. Take Netflix with you anywhere.
          </span>
        </div>
      </div>
      <div
        class="accordionstyle__Item-sc-1ev5kio-4 btjFtp"
      >
        <div
          class="accordionstyle__Header-sc-1ev5kio-5 iBTLgr"
        >
          How do I cancel?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
        <div
          class="accordionstyle__Body-sc-1ev5kio-6 jTyKSn closed"
        >
          <span>
            Netflix is flexible. There are no annoying contracts and no commitments. You can easily cancel your account online in two clicks. There are no cancellation fees – start or stop your account at any time.
          </span>
        </div>
      </div>
      <div
        class="accordionstyle__Item-sc-1ev5kio-4 btjFtp"
      >
        <div
          class="accordionstyle__Header-sc-1ev5kio-5 iBTLgr"
        >
          What can I watch on Netflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
        <div
          class="accordionstyle__Body-sc-1ev5kio-6 jTyKSn closed"
        >
          <span>
            Netflix has an extensive library of feature films, documentaries, TV programmes, anime, award-winning Netflix originals, and more. Watch as much as you want, any time you want.
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
